packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -560,6 +755,24 @@ abstract class SchedulerBinding extends BindingBase {
   int _profileFrameNumber = 0;
   final Stopwatch _profileFrameStopwatch = new Stopwatch();
   String _debugBanner;
+  bool _ignoreNextEngineDrawFrame = false;
+
+  void _handleBeginFrame(Duration rawTimeStamp) {
+    if (_warmUpFrame) {
+      assert(!_ignoreNextEngineDrawFrame);
+      _ignoreNextEngineDrawFrame = true;
```

Comment:
This looks a little fiddly

Reply:
yeah i'm not too happy with this either. I tried a bunch of different ways of handling this:
- an enum with a state machine
- hooking/unhooking the listeners
- not trying to catch this case at all
- rescheduling frames via Timer.run
- multiple bools (in this patch)
- various others

The two-bool mechanism I have here is really the least fiddly solution I could find.